GVRf/Framework/jni/objects/mesh.h

```diff
@@ -30,344 +30,322 @@
-        if (it != vec4_vectors_.end()) {
-            return it->second;
-        } else {
-            std::string error = "Mesh::getVec4Vector() : " + key + " not found";
-            throw error;
-        }
-    }
-
-    void setVec4Vector(std::string key, const std::vector<glm::vec4>& vector) {
-        vec4_vectors_[key] = vector;
-    }
-
-    Mesh* getBoundingBox();
-    const float* getBoundingBoxInfo(); // Xmin, Ymin, Zmin and Xmax, Ymax, Zmax
-
-    // /////////////////////////////////////////////////
-    //  code for vertex attribute location
-
-    void setVertexLoc(GLuint loc) {
-        vertexLoc_ = loc;
-    }
-
-    const GLuint getVertexLoc() const {
-        return vertexLoc_;
-    }
-
-    void setNormalLoc(GLuint loc) {
-        normalLoc_ = loc;
-    }
-
-    const GLuint getNormalLoc() const {
-        return normalLoc_;
-    }
-
-    void setTexCoordLoc(GLuint loc) {
-        texCoordLoc_ = loc;
-    }
-
-    const GLuint getTexCoordLoc() const {
-        return texCoordLoc_;
-    }
-
-    void setVertexAttribLocF(GLuint location, std::string key) {
-        attribute_float_keys_[location] = key;
-    }
-
-    void setVertexAttribLocV2(GLuint location, std::string key) {
-        attribute_vec2_keys_[location] = key;
-    }
-
-    void setVertexAttribLocV3(GLuint location, std::string key) {
-        attribute_vec3_keys_[location] = key;
-    }
-
-    void setVertexAttribLocV4(GLuint location, std::string key) {
-        attribute_vec4_keys_[location] = key;
-    }
-
-    // generate VAO
-    void generateVAO();
-
-    const GLuint getVAOId() const {
-        return vaoID_;
-    }
-
-    void cleanUp() {
-        std::vector<glm::vec3> vertices;
-        vertices.swap(vertices_);
-        std::vector<glm::vec3> normals;
-        normals.swap(normals_);
-        std::vector<glm::vec2> tex_coords;
-        tex_coords.swap(tex_coords_);
-        std::vector<unsigned short> triangles;
-        triangles.swap(triangles_);
-
-        if (vaoID_ != 0) {
-            glDeleteVertexArrays(1, &vaoID_);
-            vaoID_ = 0;
-        }
-
-        if (triangle_vboID_ != 0) {
-            glDeleteBuffers(1, &triangle_vboID_);
-            triangle_vboID_ = 0;
-        }
-
-        if (vert_vboID_ != 0) {
-            glDeleteBuffers(1, &vert_vboID_);
-            vert_vboID_ = 0;
-        }
-
-        if (norm_vboID_ != 0) {
-            glDeleteBuffers(1, &norm_vboID_);
-            norm_vboID_ = 0;
-        }
-
-        if (tex_vboID_ != 0) {
-            glDeleteBuffers(1, &tex_vboID_);
-            tex_vboID_ = 0;
-        }
-    }
-
-    GLuint getNumTriangles() {
-        return numTriangles_;
-    }
+	Mesh() :
+			vertices_(), normals_(), tex_coords_(), triangles_(), float_vectors_(), vec2_vectors_(), vec3_vectors_(), vec4_vectors_(), vertexLoc_(
+					-1), normalLoc_(-1), texCoordLoc_(-1), have_bounding_box_(
+					false) {
+	}
+
+	~Mesh() {
+		cleanUp();
+	}
+
+	void cleanUp() {
+		std::vector<glm::vec3> vertices;
```

Comment:
Most of the differences in this file appear to just be whitespace ... which makes it very hard to see what you really changed.